proposals/0051-function-result-sigs.rst

```diff
@@ -242,8 +242,36 @@ Alternatives
   this is neither consistent nor terribly useful.
 
 * We could detect CUSKs as we do in types to enable polymorphic recursion, but
-  this makes little sense as we are in the proccess of their deprecation.
+  this makes little sense as we are in the process of their deprecation.
 
+* We could allow contexts in the return type.
+  An earlier version of this proposal had::
+
+    funlhs -> funlhs' [:: [context =>] type]
+
+  But the usage of this can be confusing.
+  For example::
+
+    f (x :: a) :: Eq a => a
+
+  looks like it it might have type::
+
+    f :: a -> Eq a => a
+
+  Which is gratuitously higher-ranked. Base Haskell 2010 never reorders
+  arguments currently, so just moving the ``Eq a`` to the front seems overly
```

Comment:
Haskell 2010 is happy to reorder arguments like this: `f _ = abs` gets the type `f :: Num a => p -> a -> a`. Note the `Num a` moved.